### PR TITLE
Merge from CleverRaven

### PR DESCRIPTION
**Pull request by TrainTurningPoint**
```SUMMARY: Infrastructure "Merge from CleverRaven"```  

#### Purpose of change
Routine merge

### CODE REVIEW

**Comment by TrainTurningPoint** [comment]
Hi @Coolthulhu.
Could you please check that I didn't ruin the endgame lab code on merge?
There were some conflicting changes in CleverRaven repo.

**Comment by Coolthulhu** [comment]
I don't see anything wrong, although I only skimmed it because without any hints to what was in conflict, I can't really focus on any part.
Did you try generating a world, teleporting to overmap (0,0), locating the "big lab", teleporting to the finale and activating the computer to get a win screen?
If all of this works on a new world, it should be fine.

**Comment by TrainTurningPoint** [comment]
@Coolthulhu 
Yes, that's the problem. I tried and didn't find a lab with portal finale near (0, 0).
Didn't notice what could be the problem at a glance.
I'll add info about merge conflicts in a moment.

File: data/json/overmap/specials.json [hunk 1]
```diff
@@ -1,1577 +1,1513 @@
...
+      { "point": [ 1, -2, -2 ], "overmap": "central_lab" },
+      { "point": [ 2, -2, -2 ], "overmap": "central_lab" },
+      { "point": [ 1, -3, -2 ], "overmap": "central_lab_stairs" },
+      { "point": [ 1, -4, -2 ], "overmap": "central_lab_core" }
+    ],
+    "locations": [ "forest" ],
+    "city_distance": [ 20, -1 ],
+    "city_sizes": [ 0, 6 ],
+    "occurrences": [ 15, 100 ],
+    "flags": [ "UNIQUE", "ENDGAME" ],
```
Comment (TrainTurningPoint) [inline data/json/overmap/specials.json:390]: @Coolthulhu 
There were major changes in this file. 
I believe you've only added the ENDGAME flag to the "Central Lab", so I just added it over the CleverRaven changes.

File: data/json/overmap_terrain_lab.json [hunk 2]
```diff
@@ -271,10 +271,840 @@
...
     "color": "cyan",
     "see_cost": 5,
     "flags": [ "NO_ROTATE" ],
     "spawns": { "group": "GROUP_CENTRAL_LAB", "population": [ 0, 5 ], "chance": 20 }
+  },
+  {
+    "type": "overmap_terrain",
+    "id": "endgame_lab_finale",
+    "name": "portal chamber",
+    "sym": 76,
```
Comment (TrainTurningPoint) [inline data/json/overmap_terrain_lab.json]: @Coolthulhu 
Lab-related jsons was moved from overmap_terrain.json to a separate file overmap_terrain_lab.json, so I moved endgame one as well.
And I'm only just noticing that the symbol was changed as well from 76 to "L". I'll change it tomorrow to see if it helps. Maybe the entry was ignored due to changes in this field format or something.

File: src/start_location.cpp [hunk 3]
```diff
@@ -204,7 +204,7 @@ tripoint start_location::find_player_initial_location() const
     popup_nowait( _( "Please wait as we build your world" ) );
     // Spiral out from the world origin scanning for a compatible starting location,
     // creating overmaps as necessary.
-    const int radius = 32;
```
Comment (TrainTurningPoint) [inline src/start_location.cpp:207]: @Coolthulhu 
Radius got changed there.

**Comment by Coolthulhu** [comment]
>  I tried and didn't find a lab with portal finale near (0, 0).

Did you search for lab on surface? As in, using the in-game search for substring in name?
If so, that doesn't work because of central labs being weird. You have to search for access shaft (or similar name, looks like a white `0` on map) on -1 z-level.

> And I'm only just noticing that the symbol was changed as well from 76 to "L".

That might be it, though I'd expect it to fail at load, not at mapgen.

**Comment by TrainTurningPoint** [comment]
Searched for "portal chamber".
But now I've realized that the search tool only looks on the same z-level you use it, so it explains why I didn't find it.

Anyway, results are inconclusive. 
Tried to generate map twice. There was a proper central lab with access shaft and portal on the first try, but I don't see anything on the second attempt.

I think I'll merge is as it is, since it does work on occasion and we'll return to working on the endgame locations later anyway.
Thank you.

**Pull request merged**